codexctl/__init__.py

```
@@ -278,6 +305,63 @@ def version_lookup(version: str | None) -> re.Match[str] | None:
+                        print("Downgrading from 3.22+ to <3.22 requires updating the")
+                        print("bootloader on both partitions. This process will:")
+                        print("  1. Download the current version's bootloader files")
+                        print("  2. Download the target OS version")
+                        print("  3. Install the target OS version")
+                        print("  4. Update bootloader on both partitions")
+                        print("  5. Reboot")
+                        print()
+
+                        response = input("Do you want to continue? (y/n): ")
+                        if response.lower() != 'y':
+                            raise SystemExit("Installation cancelled by user")
```

Ruff (0.14.5)

Avoid specifying long messages outside the exception class

(TRY003)